exec/cnex/array.c

```diff
@@ -107,6 +123,7 @@ Array *array_copyArray(Array *a)
     for (size_t i = 0; i < r->size; i++) {
         cell_copyCell(&r->data[i], &a->data[i]);
     }
+    r->max = a->max;
```

Comment:
This seems very wrong here. You've called `array_createArrayFromSize` with the original array size, but then you're saying the max is whatever the original array had as a max. But you haven't allocated space for max in this array.

Reply:
Removed.  `max` is already set in this case, as you mentioned.